### ISSUE TITLE
[Refactor][Testing] 19 references still live on the test class

### ISSUE THREAD

**Issue by lcy-seso** [tile-ai/TileOPs#1871]
## Context

The reference computation now belongs to the narrowest shared class that names one operator — normally the workload — because it is the executable form of the manifest's `ref_api` and not a decision either consumer makes. See `docs/design/trust-model.md` §Workloads layer.

64 references moved onto their workload. **19 did not**, and are listed here with what specifically blocked each. A workload that describes only an input shape is out of scope by the rule and is not counted.

### A. One test class serves several ops through an `op_kind` string — 7

`ReduceTest`(`SumWorkload`), `WelfordTest`(`StdWorkload`), `WelfordNonAlignedTest`(`RandnWorkload`), `ArgreduceTest`(`ArgmaxWorkload`), `LogicalReduceTest`(`AnyWorkload`), `VectorNormTest`(`L1NormWorkload`), `CumulativeTest`(`CumulativeWorkload`).

```python
class ReduceTest(SumWorkload, TestBase):
    def __init__(self, m, n, dtype, op_kind: str): ...
    def ref_program(self, x):
        if self.op_kind == "sum":  return x_f32.sum(dim=-1)...
        elif self.op_kind == "mean": ...
```

The workload is named for one op and used for a family. The reference cannot move onto `SumWorkload` without teaching that class about `mean`, `max` and the rest.

### B. The reference is injected as a callable — 3

`BitwiseTest`(`BitwiseWorkload`), `LogicalTest`(`LogicalWorkload`), `SpecialTest`(`SpecialWorkload`).

```python
class BitwiseTest(BitwiseWorkload, TestBase):
    def __init__(self, n_total, ref_fn): self.ref_fn = ref_fn
    def ref_program(self, a, b): return self.ref_fn(a, b)
```

**Same defect as A**, expressed as a lambda rather than a string.

### C. The reference reads a parameter the workload does not model — 3

`SoftmaxTest`, `LogSoftmaxTest`, `LogSumExpTest`.

```python
class SoftmaxTest(SoftmaxWorkload, TestBase):
    def __init__(self, shape, dtype, dim: int = -1): self.dim = dim
    def ref_program(self, x): return F.softmax(x.float(), dim=self.dim)...
```

`dim` is a semantic parameter — the manifest declares it — but `SoftmaxWorkload` is `(shape, dtype)`.

### D. The reference rests on module-level helpers — 1

`RopeTest`. `ref_program` sits on ~200 lines of frequency builders in the test file (`_compute_freqs_cis_base`, `ref_rope_neox`, the yarn / longrope / llama31 variants), half of them called directly by other tests in the same file.

### E. GQA — 5

`GroupedQueryAttentionFwdTest`, `…BwdTest`, `…DecodePagedTest`, `…SlidingWindowFwdTest`, `…SlidingWindowVarlenFwdTest`.

- fwd: `test_gqa.py` also holds `_gqa_prefill_ref` and `_gqa_prefill_varlen_ref`, for which no workload class exists. Moving one of three splits the family's references across two layers.
- bwd: the reference runs `output.backward(grad_output)` and returns `q.grad` — it depends on `gen_inputs` having built tensors with `requires_grad_`, a contract nothing expresses.
- decode-paged: the benchmark baseline deliberately differs (`sdpa_kernel(MATH)`), so nothing would be deduplicated; the reference also encodes the page-table layout.
- sliding-window ×2: would move cleanly, but no benchmark copy exists, so the move deduplicates nothing today.

## Suggested refactor

**A and B are one problem: a workload named for an op but scoped to a family.** Two honest resolutions, and the choice is the same for all ten:

1. **Split into per-op workloads** — `SumWorkload`, `MeanWorkload`, `MaxWorkload`, each carrying its own reference. Matches the manifest, which already has one entry per op. Costs one small class per op.
2. **Rename to what they are** — `Reduce2DWorkload`, `ElementwisePairWorkload` — at which point they fall under the shape-only rule and the reference legitimately stays with the consumer.

Prefer 1 where a benchmark also needs the reference (duplication is real today), 2 where only the test consumes it.

**C is the cheapest and clearly right**: `dim` belongs on `SoftmaxWorkload` — it shapes what the op computes and the manifest already declares it. Move the parameter, and the reference moves with it.

**D**: move the frequency builders into the workload module as the reference's implementation and have the other tests import them from there. Promotes about seven helpers to a public surface; worth doing only alongside a look at whether those tests should be exercising the builders directly at all.

**E**: leave until GQA is refactored on its own terms. Two of the five are only waiting on a workload class for the prefill variants; one is waiting on `gen_inputs`/`ref_program` agreeing about `requires_grad`; two would move today but buy nothing. Nothing in this list should shape that refactor — it is the other way round.

## Not urgent

Nothing here is broken. Each remaining reference has exactly one consumer, so there is no second copy to drift. The cost is that a benchmark wanting a torch baseline for any of these ops has to write the reference a second time — which is how the 313 duplicated lines accumulated in the first place.

## Reproduce

```
python - <<'PY'
import ast, pathlib
SHAPE = {"RandnFlatWorkload","RandnPairWorkload","PositivePairWorkload",
         "GatedRandnWorkload","AdaptivePool2dWorkload"}
for p in sorted(pathlib.Path("tests").rglob("*.py")):
    for n in ast.walk(ast.parse(p.read_text())):
        if not isinstance(n, ast.ClassDef): continue
        if not any(isinstance(m, ast.FunctionDef) and m.name == "ref_program" for m in n.body): continue
        bases = [b.id for b in n.bases if isinstance(b, ast.Name)]
        if len(bases) == 2 and bases[1] == "TestBase" and bases[0] not in SHAPE:
            print(f"{n.name} <- {bases[0]}  ({p})")
PY
```
